src/ray/object_manager/object_directory.h

```diff
@@ -99,6 +99,13 @@ class ObjectDirectoryInterface {
   /// \return Status of whether this method succeeded.
   virtual ray::Status ReportObjectRemoved(const ObjectID &object_id,
                                           const ClientID &client_id) = 0;
+
+  /// Go through all the client information.
+  ///
+  /// \param success_cb A callback which handles the success of this method.
+  ///                   This function will be called multiple times.
+  /// \return Status of whether this asynchronous request succeeded.
+  virtual ray::Status GetAllClientInfo(const InfoSuccessCallback &success_cb) = 0;
```

Comment:
Also, this function always succeeds right? So maybe it should just return `Void`?